newscript.go

```diff
@@ -0,0 +1,404 @@
+
+func goCompatibility() error {
+	fmt.Printf("--------Checking Go Compatibility \n")
+	cmd := exec.Command("go", "version")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+	// The string is originally in the form: go version go1.9rc2_cl165246139 linux/amd64 where 1.9 is the go version
+	termString, err := strconv.ParseFloat(strings.Split(out.String(), " ")[2][2:5], 64)
+	if err != nil {
+		return err
+	}
+	if termString > 1.7 {
+		fmt.Println("Compatible go version")
+	} else {
+		return errors.New("Please install go v1.7 or greater.")
+	}
+	return nil
+}
+
+func goGet() error {
+	fmt.Printf("--------Getting u-root \n")
+	cmd1 := exec.Command("go", "get", "github.com/u-root/u-root")
+	err := cmd1.Run()
+	//TODO: Figure why this ^ isn't working
+	/*if err != nil {
+		return err
+	}*/
+	fmt.Printf("--------Gotu-root \n")
+	gopath := fmt.Sprintf("GOPATH=%s/go", homeDir)
+	bbpath := fmt.Sprintf("GOPATH=%s/go/src/github.com/u-root/u-root/bb/bb", homeDir)
+	cmd3 := exec.Command(gopath, bbpath)
+	err = cmd3.Run()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("--------Getting bb \n")
+/* Removing in favor of bb
+	cmd4 := exec.Command("go", "run", "scripts/ramfs.go")
+	err = cmd4.Run()
+	if err != nil {
+		return err
+	}
+*/
+	if _, err := os.Stat("/tmp/initramgs.linux_amd64.cpio"); err != nil {
+		return err
+	}
+	fmt.Printf("Created the initramfs in /tmp/")
+	return nil
+}
+
+// Get the right kernel for the current config from github (needs work)
+// https://github.com/LaconicPneumonic/linux-4.12.7.git
+func kernelGet() error {
+	fmt.Printf("-------- Getting the kernel \n")
+	cmd8 := exec.Command("git", "clone", "https://github.com/LaconicPneumonic/linux-4.12.7.git")
+	err := cmd8.Run()
+	if err != nil {
+		fmt.Printf("didn't clone kernel from Anthony's repo")
+		return err
+	}
+	return nil
+}
+
+
+
+// Get the right kernel for the current config in NiChrome (Needs work)
+func linuxKernelGet() error {
+	fmt.Printf("-------- Getting the kernel \n")
+	fmt.Printf("Current version of Kernel is %v\n", kernelVersion)
+	name := fmt.Sprintf("%s.tar", linuxVersion)
+	out, err := os.Create(name)
+	fmt.Printf("Kernel saved in %s\n", name)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	linkName := fmt.Sprintf("%s%s%s", "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-", kernelVersion, ".tar.xz")
+	fmt.Printf("Pulling from link %s\n", linkName)
+	resp, err := http.Get(linkName)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
+	}
+	cmd5 := exec.Command("tar", "xf", fmt.Sprintf("%s.tar",linuxVersion))
+	err = cmd5.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// To get the newest Kernel from kernel.org
+func newestkernelGet() error {
+	jsonFile, err := http.Get("http://www.kernel.org/releases.json")
+	if err != nil {
+		return err
+	}
+	defer jsonFile.Body.Close()
+	d, err := ioutil.ReadAll(jsonFile.Body)
```

Comment:
This is not as neat ;-)
you should use the encoding/json package to decode the json to a struct. The way you're doing this is fragile. We can get this later.